stablehlo/dialect/CheckOps.td

```diff
@@ -0,0 +1,132 @@
+/* Copyright 2023 The TensorFlow Authors. All Rights Reserved.
+   Copyright 2023 The StableHLO Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef STABLEHLO_DIALECT_CHECK_OPS
+#define STABLEHLO_DIALECT_CHECK_OPS
+
+include "stablehlo/dialect/Base.td"
+
+//===----------------------------------------------------------------------===//
+// CHECK dialect.
+//===----------------------------------------------------------------------===//
+
+def CHECK_Dialect : Dialect {
+  let name = "check";
+  let cppNamespace = "::mlir::stablehlo::check";
+
+  let description = [{
+    Opset to implement test assertions for StableHLO interpreter.
+  }];
+
+  let useDefaultAttributePrinterParser = 0;
+  let useDefaultTypePrinterParser = 0;
+  let useFoldAPI = kEmitFoldAdaptorFolder;
+}
+
+//===----------------------------------------------------------------------===//
+// CHECK op definitions.
+//===----------------------------------------------------------------------===//
+
+def CHECK_ExpectEqOp : Op<CHECK_Dialect, "expect_eq", [SameTypeOperands]> {
+  let summary = [{Checks that the tensor operands are equal}];
+  let description = [{
+    Verifies that the operands are exactly equal. For floating-point and complex
+    types, uses bitwise comparison for equality.
+
+    Issues a non-fatal failure if the verification fails.
+
+    ```mlir
+    check.expect_eq(%arg0, %arg1) : tensor<5xi32>
+    ```
+  }];
+
+  let arguments = (ins
+      HLO_Tensor:$lhs,
+      HLO_Tensor:$rhs
+  );
+
+  let assemblyFormat = "`(` $lhs `,` $rhs `)` attr-dict `:` type($lhs)";
+}
+
+def CHECK_ExpectEqConstOp :
+    Op<CHECK_Dialect, "expect_eq_const", [AllTypesMatch<["lhs", "value"]>]> {
+  let summary = [{Checks that the tensor operand is equal to some constant}];
+  let description =  [{
+    Verifies that the tensor operand is exactly equal to a constant attribute.
+    For floating-point and complex types, uses bitwise comparison for equality.
+
+    Issues a non-fatal failure if the verification fails.
+
+    ```mlir
+    check.expect_eq_const(%arg0, dense<[1, 2]> : tensor<2xi32>) : tensor<2xi32>
```

Comment:
The syntax looks kind of weird at the moment, with the same type duplicated twice. Could this be something like `check.expect_eq %arg0, dense<[1, 2]> : tensor<2xi32>`? (Without `_const` to make things more brief).

Reply:
done.